auth-web/src/components/datatable/BaseVDataTable.vue

```diff
@@ -0,0 +1,277 @@
+        <slot name="header-filter-slot" :headers="headers">
+          <tr :style="{ 'background-color': headerBg }">
+            <th
+              v-for="header in headers"
+              :key="header.col"
+              :class="[header.class, 'base-table__header__filter pb-5']"
+            >
+              <slot :name="'header-filter-slot-' + header.col" :header="header">
+                <v-select
+                  v-if="header.hasFilter && header.customFilter.type === 'select'"
+                  :class="['base-table__header__filter__select']"
+                  :clearable="header.customFilter.clearable"
+                  dense
+                  filled
+                  hide-details
+                  item-text="text"
+                  item-value="value"
+                  :items="header.customFilter.items"
+                  :label="!header.customFilter.value ? header.customFilter.label || '' : ''"
+                  :open-on-clear="true"
+                  v-model="header.customFilter.value"
+                  @reset="filter(header)"
+                  @input="filter(header)"
+                />
+                <v-text-field
+                  v-else-if="header.hasFilter && header.customFilter.type === 'text'"
+                  :class="['base-table__header__filter__textbox', header.customFilter.value ? 'active' : '']"
+                  :clearable="header.customFilter.clearable"
+                  dense
+                  filled
+                  hide-details
+                  :placeholder="!header.customFilter.value ? header.customFilter.label || '' : ''"
+                  v-model="header.customFilter.value"
+                  @reset="filter(header)"
+                  @input="filter(header)"
+                />
+              </slot>
+            </th>
+          </tr>
+        </slot>
+      </thead>
+    </template>
+
+    <!-- Items -->
+    <template v-slot:item="{ item }">
+      <tr :key="item[itemKey]">
+        <td
+          v-for="header in headers" :key="'item-' + header.col"
+          :class="[header.itemClass, 'base-table__item']"
+        >
+          <slot :header="header" :item="item" :name="'item-slot-' + header.col">
+            <span v-if="header.itemFn" v-html="header.itemFn(item[header.col])" />
+            <span v-else>{{ item[header.col] }}</span>
+          </slot>
+        </td>
+      </tr>
+    </template>
+
+    <!-- Loading -->
+    <template v-slot:loading>
+      <div class="py-8 base-table__text" v-html="loadingText" />
+    </template>
+
+    <!-- No data -->
+    <template v-slot:no-data>
+      <div class="py-8 base-table__text" v-html="noDataText" />
+    </template>
+  </v-data-table>
+</template>
+
+<script lang="ts">
+import { computed, defineComponent, reactive, ref, toRefs, watch } from '@vue/composition-api'
+import { BaseTableHeaderI } from './interfaces'
+import { DEFAULT_DATA_OPTIONS } from './resources'
+import { DataOptions } from 'vuetify'
+import _ from 'lodash'
```

Comment:
`import debounce from 'lodash/debounce'` if possible?

Reply:
We might use lodash for cloneDeep? but I think there is a new built in method?

Reply:
https://web.dev/structured-clone/

Reply:
ooh okay yeah I'll check that out

Reply:
I think I'd rather keep it the way it is. 'structuredClone' is not in Node 16 which is what we're in for auth-web. Sound good?

Reply:
Sounds good to me, I didn't realize it wasn't in Node 16, we should turf it when upgrade Node though.